src/baldr/graphreader.cc

```diff
@@ -36,22 +42,59 @@ tile_gone_error_t::tile_gone_error_t(std::string prefix, baldr::GraphId edgeid)
 }
 
 GraphReader::tile_extract_t::tile_extract_t(const boost::property_tree::ptree& pt) {
+  // A lambda for loading the contents of a graph tile tar from an index file
+  auto index_loader = [this](const std::string& filename, const char* index_begin,
+                             const char* file_begin, size_t size,
+                             tar::TarType tar_type) -> decltype(midgard::tar::contents) {
+    // has to be our specially named index.bin file
+    if (filename != "index.bin")
+      return {};
+
+    // get the info
+    decltype(midgard::tar::contents) contents;
+    auto entry_size = sizeof(tile_index_entry);
+    auto entries = midgard::iterable_t<tile_index_entry>(reinterpret_cast<tile_index_entry*>(
+                                                             const_cast<char*>(index_begin)),
+                                                         size / sizeof(tile_index_entry));
+    for (const auto& entry : entries) {
+      auto inserted = contents.insert(
+          std::make_pair(std::to_string(entry.tile_id),
+                         std::make_pair(const_cast<char*>(file_begin + entry.offset), entry.size)));
+      if (tar_type == tar::TarType::kTiles) {
+        tiles.emplace(std::piecewise_construct, std::forward_as_tuple(entry.tile_id),
+                      std::forward_as_tuple(const_cast<char*>(file_begin + entry.offset),
+                                            entry.size));
+      } else if (tar_type == tar::TarType::kTraffic) {
+        traffic_tiles.emplace(std::piecewise_construct, std::forward_as_tuple(entry.tile_id),
+                              std::forward_as_tuple(const_cast<char*>(file_begin + entry.offset),
+                                                    entry.size));
```

Comment:
a new `TarType` enum which can be passed to the `tar` constructor and further to this lambda so we can decide which container to write to